### ISSUE TITLE
 go  get github.com/mattn/go-sqlite3 error!

### ISSUE THREAD

**Issue by acekingke** [mattn/go-sqlite3#199]
$ go  get github.com/mattn/go-sqlite3
# github.com/mattn/go-sqlite3
<command-line>:0:14: error: conflicting types for 'localtime'
In file included from c:\go\3party\src\github.com\mattn\go-sqlite3\sqlite3-bindi
ng.c:14589:0:
c:\mingw\include\time.h:122:45: note: previous declaration of 'localtime' was he
re
 _CRTIMP struct tm*  __cdecl __MINGW_NOTHROW localtime (const time_t*);
                                             ^
c:\go\3party\src\github.com\mattn\go-sqlite3\sqlite3-binding.c: In function 'osL
ocaltime':
c:\go\3party\src\github.com\mattn\go-sqlite3\sqlite3-binding.c:14983:3: warning:
 passing argument 1 of 'localtime' from incompatible pointer type [enabled by de
fault]
   pX = localtime(t);
   ^
<command-line>:0:14: note: expected 'const __time32_t *' but argument is of type
 'time_t *'

The  version of gcc is

$ gcc -v
Using built-in specs.
COLLECT_GCC=c:\mingw\bin\gcc.exe
COLLECT_LTO_WRAPPER=c:/mingw/bin/../libexec/gcc/mingw32/4.8.1/lto-wrapper.exe
Target: mingw32
Configured with: ../gcc-4.8.1/configure --prefix=/mingw --host=mingw32 --build=m
ingw32 --without-pic --enable-shared --enable-static --with-gnu-ld --enable-lto
--enable-libssp --disable-multilib --enable-languages=c,c++,fortran,objc,obj-c++
,ada --disable-sjlj-exceptions --with-dwarf2 --disable-win32-registry --enable-l
ibstdcxx-debug --enable-version-specific-runtime-libs --with-gmp=/usr/src/pkg/gm
p-5.1.2-1-mingw32-src/bld --with-mpc=/usr/src/pkg/mpc-1.0.1-1-mingw32-src/bld --
with-mpfr= --with-system-zlib --with-gnu-as --enable-decimal-float=yes --enable-
libgomp --enable-threads --with-libiconv-prefix=/mingw32 --with-libintl-prefix=/
mingw --disable-bootstrap LDFLAGS=-s CFLAGS=-D_USE_32BIT_TIME_T
Thread model: win32
gcc version 4.8.1 (GCC)

**Comment by mattn** [comment]
Windows 32bit? 64bit?

**Comment by acekingke** [comment]
The mingw is 32bit, the system is 64bit

**Comment by mattn** [comment]
Could you please try workaround like below?

https://github.com/mattn/go-sqlite3/blob/master/sqlite3_windows.go#L11

replace 
```
#cgo windows,386 CFLAGS: -D_localtime32=localtime
```
to
```
#cgo windows,386 CFLAGS: -D_localtime32=localtime -D_localtime64=localtime
```

**Comment by acekingke** [comment]
I have replace it ,but it don't work 
![image](https://cloud.githubusercontent.com/assets/7586502/7407447/b89d5f30-ef44-11e4-937f-eecf880e33c3.png)
and then
![image](https://cloud.githubusercontent.com/assets/7586502/7407458/d832e900-ef44-11e4-9b76-45d6b2e391ff.png)

**Comment by niniwzw** [comment]
https://github.com/mattn/go-sqlite3/blob/master/sqlite3_windows.go#L11

replace

#cgo windows,386 CFLAGS: -D_localtime32=localtime

to

#cgo windows,386 CFLAGS: -D_USE_32BIT_TIME_T

is ok

**Comment by mattn** [comment]
@acekingke could you please try out @niniwzw 's suggestion?

**Comment by taruti** [comment]
Stumbled upon this issue. It occurs at least with mingw32 gcc-4.8.1.

```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=c:/mingw/bin/../libexec/gcc/mingw32/4.8.1/lto-wrapper.exe
Target: mingw32
Configured with: ../gcc-4.8.1/configure --prefix=/mingw --host=mingw32 --build=mingw32 --without-pic --enable-shared --enable-static --with-gnu-ld --enable-lto --enable-libssp --disable-multilib --enable-languages=c,c++,fortran,objc,obj-c++,ada --disable-sjlj-exceptions --with-dwarf2 --disable-win32-registry --enable-libstdcxx-debug --enable-version-specific-runtime-libs --with-gmp=/usr/src/pkg/gmp-5.1.2-1-mingw32-src/bld --with-mpc=/usr/src/pkg/mpc-1.0.1-1-mingw32-src/bld --with-mpfr= --with-system-zlib --with-gnu-as --enable-decimal-float=yes --enable-libgomp --enable-threads --with-libiconv-prefix=/mingw32 --with-libintl-prefix=/mingw --disable-bootstrap LDFLAGS=-s CFLAGS=-D_USE_32BIT_TIME_T
Thread model: win32
gcc version 4.8.1 (GCC) 
```


The suggested
```
#cgo windows,386 CFLAGS: -D_USE_32BIT_TIME_T
```
worked for that tool chain.

**Comment by mattn** [comment]
Closed by #262
